### ISSUE TITLE
Rumba sometimes does not work due to a bug in Dyalog with dfns

### ISSUE THREAD

**Issue by aplteam** [the-carlisle-group/Rumba#56]
According to Dyalog there is a general problem with dfns when threads are used. It's a difficult bug to fix but there is a workaround: convert two functions from dfns to tfns and it appears to work.

The functions in question are `DefaultResponseHeaders` and `HandleRequest`. The bug manisfests itself by producing an impossible VALUE ERROR:

```
3:VALUE ERROR: Undefined name: ce
DefaultResponseHeaders[7] h,←(~ce)/⊂'Content-Length'(⍕⍵.ContentLength)
```

In order to reach line 7 line 3 must have been executed, and that defines `ce`, which is why I cioned it `impossible VALUE ERROR`.

This is Mantis 19300.

In order to fix the problem I've changed the two original dfns to these tfns:

```
 response←DefaultResponseHeaders response;ce;h;b
 ce←response.Chunked
 h←⊂'Date'(GetCurrentUTCHttpDate'')
 h,←⊂'Server' 'Rumba'
 h,←response.Close/⊂'Connection' 'close'
 h,←(~ce)/⊂'Content-Length'(⍕response.ContentLength)
 h,←ce/⊂'Transfer-Encoding' 'chunked'
 b←(response.ContentLength>0)∧~response.Chunked
 h,←b/⊂'Content-Type'response.ContentType
 h,←(0<≢response.Location)/⊂'Location'response.Location
 h,←(0<≢response.ETag)/⊂'ETag'response.ETag
 h,←(0≠response.LastModified)/⊂'Last-Modified'(FloatToHTTPDate response.LastModified)
 response.DefaultHeaders←h
```

```
 r←HandleRequest m;c;s
 r←0
 :If m≠0
     c←m.Connection
     s←c.Server
     m.Tid←⎕TID
     r←ExecuteRequestHandler m
     r.Close←~IsPersistent m
     {}SendResponse r
     m.ResponseSent←1
     c.MessageLog,←⊂m.Method m.RequestTarget r.StatusCode
     {}s LogHTTPToSession r
     {}UpdateStats r
     {}Log r
     {}DisplayRequestResponse r
     :If r.Close
         ExpungeConnection c
     :Else
         c.Messages←1↓c.Messages
     :EndIf
 :EndIf
```

**Comment by aplteam** [comment]
19300 is supposed to be fixed in 19.0, but the fix won't be backported to earlier versions.

**Comment by PaulMansour** [comment]
Done.
